### PR TITLE
Verify that nothing but tables exist in base table

### PR DESCRIPTION
**Pull request by TheNeikos**

### CODE REVIEW

**Comment by TheNeikos** [comment]
RFR

File: libimagstore/src/store.rs [hunk 1]
```diff
@@ -265,8 +267,9 @@ impl<'a> ::std::ops::DerefMut for FileLockEntry<'a> {
 }
 
 impl<'a> Drop for FileLockEntry<'a> {
+    /// This will silently ignore errors, use `Store::update` if you want to catch the errors
     fn drop(&mut self) {
-        self.store._update(self).unwrap()
+        let _ = self.store._update(self);
```
Comment (matthiasbeyer) [inline libimagstore/src/store.rs]: For documentation: We argued and we came to the conclusion that this is **valid**.

File: libimagstore/src/store.rs [hunk 2]
```diff
@@ -352,17 +349,33 @@ fn build_default_header() -> BTreeMap<String, Value> {
...
+    if let Err(e) = verify_header(&t) {
+        Err(ParserError::new(ParserErrorKind::HeaderInconsistency, None))
     } else {
         Ok(t)
     }
 }
 
+fn has_only_tables(t: &Table) -> bool {
+    debug!("Verifying that table has only tables");
+    !t.iter().any(|(_, x)| if let &Value::Table(_) = x { false } else { true })
```
Comment (matthiasbeyer) [inline libimagstore/src/store.rs]: Please use `Iterator::all()` that makes that less of a brainfuck here...

**Comment by matthiasbeyer** [comment]
We tested this locally.

**Pull request merged**